networking/src/NTP/Client.hs

```diff
@@ -39,16 +38,19 @@ import           System.Wlog (LoggerName, WithLogger, logDebug, logError, logInf
                               modifyLoggerName)
 
 import           Mockable.Class (Mockable)
-import           Mockable.Concurrent (Async, Concurrently, Delay, concurrently, forConcurrently,
+import           Mockable (Async, Concurrently, CurrentTime, Delay, concurrently, currentTime, forConcurrently,
                                       timeout, withAsync)
 import           NTP.Packet (NtpPacket (..), evalClockOffset, mkCliNtpPacket, ntpPacketSize)
 import           NTP.Util (createAndBindSock, resolveNtpHost, selectIPv4, selectIPv6,
                            udpLocalAddresses, withSocketsDoLifted)
 
-data NtpClientSettings m = NtpClientSettings
+data NtpStatus = NtpSyncOk | NtpDesync Microsecond
+    deriving (Eq, Show)
+
+data NtpClientSettings = NtpClientSettings
```

Comment:
What is the difference between `NtpClientSettings` and `NtpConfiguration`? Seems weird that we have a `TVar` in a "settings" record.